library/ssl_tls.c

```diff
@@ -5579,14 +5579,7 @@ void mbedtls_ssl_handshake_free( mbedtls_ssl_context *ssl )
         mbedtls_free( (void*) handshake->sig_algs );
     handshake->sig_algs = NULL;
 #endif /* MBEDTLS_DEPRECATED_REMOVED */
-#if defined(MBEDTLS_SSL_PROTO_TLS1_3)
-    if( ssl->handshake->certificate_request_context )
-    {
-        mbedtls_free( (void*) handshake->certificate_request_context );
-    }
-    handshake->certificate_request_context = NULL;
```

Comment:
Those two last lines can be removed as the whole structure is zeroed at the end of the function but we must keep the call to `mbedtls_free` to free the allocated context.